### PR TITLE
Feature/tr 294 save item state on pause

### PR DESCRIPTION
**Pull request by meefox**
Related to https://oat-sa.atlassian.net/browse/TR-293
when you click on the pause button you can see the response message (you needed to preserve log) that contains 
extra data about the item
Screenshot 2021-03-03 at 10 25 15

### CODE REVIEW

File: src/provider/qti.js [hunk 1]
```diff
@@ -487,10 +487,14 @@ var qtiProvider = {
                 }
             })
             .on('pause', function(data) {
+                const context = self.getTestContext();
```
Comment (btamas) [inline src/provider/qti.js:490]: Can you please call it `testContext` because this is the convention for it.

File: src/provider/qti.js [hunk 2]
```diff
@@ -487,10 +487,14 @@ var qtiProvider = {
...
+
                 this.setState('closedOrSuspended', true);
 
                 this.getProxy()
                     .callTestAction('pause', {
+                        itemDefinition: context.itemIdentifier,
```
Comment (btamas) [inline src/provider/qti.js:496]: Can you please call it `itemIdentifier`, because it is the convention and it is just an id, not the definition itself.

Comment (meefox) [reply src/provider/qti.js:496]: this was a request from @emgolubev to call it like this

Comment (jsconan) [reply src/provider/qti.js]: Oh yes right! This is the name of the API parameter. Good catch.

**Review by jsconan — APPROVED** [review]
Good job! Working like a charm!

Note: unit tests are still failing, but this is not related to this PR.

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

File: src/provider/qti.js (hunk 2 above)
Comment (meefox) [reply src/provider/qti.js]: please be sync with https://github.com/oat-sa/extension-tao-testqti/pull/1986/files

Comment (emgolubev) [reply src/provider/qti.js]: let's keep `itemDefinition` please

**Comment by jsconan** [comment]
BTW, the ticket number in the branch name, as well as in the PR title, is not correct, it should be TR-293 instead of TR-294.

**Review by btamas — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by lecosson — APPROVED** [review]
works as exected 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

image

**Pull request merged**